### DISCUSSION TITLE
Stacking min and max breakpoints doesn’t work

### DISCUSSION THREAD

**Discussion by kokoshneta** [tailwindlabs/tailwindcss#10612] [Help] [answered]
I have Tailwind installed and updated to v3.2.7 (just updated with `npm install -D tailwindcss@latest postcss@latest autoprefixer@latest` to be sure, and `npm info tailwindcss version` shows 3.2.7), so I should have access to the [out-of-the-box `max-*` breakpoints](https://github.com/tailwindlabs/tailwindcss/pull/9558). I’m using Tailwind in a Laravel project, and this is on my development machine (MacBook Pro M1, macOS Ventura, serving Laravel through Valet).

I have my screens extended like this:

```
module.exports = {
  theme: {
    extend: {
      screens: {
        'xs': '500px',
        '3xl': '1681px',
        '4xl': '1901px'
      }
    }
  }
}
```

These breakpoints all work as intended when used alone:

```
<div class="hidden xs:block">Shown on xs and above</div>
<div class="max-xs:hidden">Hidden below xs</div>
```

However, when trying to stack breakpoints to target a range, the class is never generated at all, so this does nothing:

```
<div class="xs:max-sm:hidden">Should be hidden between xs and sm, but is always shown</div>
```

With that, there is no `@media not all` query in the generated CSS file, not does `max-sm` appear anywhere in the generated CSS file.

Given that it works just fine [in this Codepen](https://codepen.io/kokoshneta/pen/qBMdorE), I assume this is an issue with my setup, somehow. But how?

Apart from extended screen breakpoints (which the Codepen also has), what might be some possible factors that cause these stacked breakpoints not to work?

**Comment by adamwathan** [comment, marked as answer]
Hey! Definitely seems to work here:

https://play.tailwindcss.com/ksD6ZpTm0T?size=562x720&file=config

Can you create a reproduction? No way to tell you what to change in your setup without seeing your setup.

**Reply by kokoshneta** [reply]
Just figured it out… turned out to be me just being dumb. 🙄

I’ve got Laravel Mix running in watch mode (`npm run watch`), and I hadn’t restarted that process since before I updated Tailwind, so obviously it wasn’t picking up on any changes added since the last time I restarted the watch process.

A simple restart of that, and everything works as it should.
